app/src/main/java/org/wikipedia/edit/preview/EditPreviewFragment.kt

```diff
@@ -0,0 +1,301 @@
+            if (otherTag.selected) {
+                otherTag.isSelected = false
+            } else {
+                (requireActivity() as EditSectionActivity).showCustomSummary()
+            }
+        }
+
+        if (savedInstanceState != null) {
+            for (i in summaryTags.indices) {
+                summaryTags[i].isSelected = savedInstanceState.getBoolean("summaryTag$i", false)
+            }
+            if (savedInstanceState.containsKey("otherTag")) {
+                otherTag.isSelected = true
+                otherTag.text = savedInstanceState.getString("otherTag")
+            }
+        }
+        binding.editPreviewContainer.visibility = View.GONE
+        return binding.root
+    }
+
+    fun setCustomSummary(summary: String) {
+        otherTag.text = if (summary.isNotEmpty()) summary else getString(R.string.edit_summary_tag_other)
+        otherTag.isSelected = summary.isNotEmpty()
+    }
+
+    /**
+     * Fetches preview html from the modified wikitext text, and shows (fades in) the Preview fragment,
+     * which includes edit summary tags. When the fade-in completes, the state of the
+     * actionbar button(s) is updated, and the preview is shown.
+     * @param title The PageTitle associated with the text being modified.
+     * @param wikiText The text of the section to be shown in the Preview.
+     */
+    fun showPreview(title: PageTitle, wikiText: String) {
+        DeviceUtil.hideSoftKeyboard(requireActivity())
+        (requireActivity() as EditSectionActivity).showProgressBar(true)
+        val url = ServiceFactory.getRestBasePath(model.title!!.wikiSite) +
+                RestService.PAGE_HTML_PREVIEW_ENDPOINT + UriUtil.encodeURL(title.prefixedText)
+        val postData = "wikitext=" + UriUtil.encodeURL(wikiText)
+        binding.editPreviewWebview.postUrl(url, postData.toByteArray())
+        ViewAnimations.fadeIn(binding.editPreviewContainer) { requireActivity().invalidateOptionsMenu() }
+        ViewAnimations.fadeOut(requireActivity().findViewById(R.id.edit_section_container))
+    }
+
+    private fun initWebView() {
+        binding.editPreviewWebview.webViewClient = object : OkHttpWebViewClient() {
+            override fun getModel(): PageViewModel {
+                return this@EditPreviewFragment.model
+            }
+
+            override fun getLinkHandler(): LinkHandler {
+                return this@EditPreviewFragment.linkHandler
+            }
+
+            override fun onPageFinished(view: WebView, url: String) {
+                super.onPageFinished(view, url)
+                if (!isAdded) {
+                    return
+                }
+                (requireActivity() as EditSectionActivity).showProgressBar(false)
+                requireActivity().invalidateOptionsMenu()
+                bridge.execute(JavaScriptActionHandler.setTopMargin(0))
+            }
+        }
+
+        bridge.addListener("setup", object : JSEventListener {
```

Comment:
The `interface JSEventListener` can be `fun interface JSEventListener`.

Reply:
correct!